src/rmp/src/Restructure.cpp

```diff
@@ -103,7 +105,8 @@ void Restructure::run(char* liberty_file_name,
   work_dir_name_ = workdir_name;
   work_dir_name_ = work_dir_name_ + "/";
 
-  removeConstCells();
+  if (opt_mode_ <= Mode::AREA_3) // Only in area mode
```

Comment:
Introduce a isAreaMode helper rather than repeat this test.  If you add an AREA_4 multiple places will need updating.

Reply:
Added, it helps. Thanks.